libpod/boltdb_state.go

```diff
@@ -504,18 +504,28 @@ func (s *BoltState) Container(id string) (*Container, error) {
 	}
 	defer s.deferredCloseDBCon(db)
 
+	var rewrite bool
```

Comment:
I am not sure about this, we can only run it for the root user and only for distros we control. I would like to keep old container working without manual intervention.
I was wondering if we can instead of writing the config back to the db just convert every time time we read it. Converting ports from old to the new format is fast, see `Benchmark_ocicniPortsToNetTypesPorts` results.

Reply:
We already have to read all containers as part of a state refresh, so it wouldn't really be a net loss. The downside is that we'd either have to force a reboot after installing the update (else ports would not work) or convince refresh to be a bit more selective about how it wipes state information.

Reply:
@Luap99 How about doing a temporary conversion as you suggesting, keeping 387-396, and then letting Refresh do a rewrite of the state, so a reboot finalizes the changes in the DB?

Reply:
@mheon SGTM